Compiler/BackEnd/Tearing.mo

```diff
@@ -2911,8 +2911,15 @@ algorithm
   if debug then execStat("Tearing.ModifiedCellierHeuristic_3 - 2"); end if;
 
   // 2. Determine the variables in causEq
-  msel := Array.select(mIn,causEq);
-  potentialTVars := List.unique(List.flatten(arrayList(msel)));
+  //msel := Array.select(mIn,causEq);
+
+  potentialTVars := {};
+  for e in causEq loop
+    potentialTVars := listAppend(arrayGet(mIn, e), potentialTVars);
+  end for;
+  potentialTVars := List.unique(potentialTVars);
```

Comment:
I guess it would. A node in a tree costs 8 words (2 words is overhead due to GC; a leaf costs 4 words, 1 word overhead), while a cons-cell costs 4 words (1 word overhead).